### ISSUE TITLE
`clippy::question_mark` ignores `#[cfg]`-ed out statements in `let-else`

### ISSUE THREAD

**Issue by GoldsteinE** [rust-lang/rust-clippy#13642]
### Summary

`else` block in `let-else` can have different behaviour depending on build configuration. Clippy ignores it and suggests replacing `let-else` block with `?`, completely removing `#[cfg]`-ed out logic.

As Clippy can’t look inside `#[cfg]`-ed out statements, probably it shouldn’t ever suggest removing such code.

### Lint Name

clippy::question_mark

### Reproducer

I tried this code:

```rust
fn returns_option() -> Option<u8> {
    None
}

pub fn check() -> Option<u8> {
    let Some(x) = returns_option() else {
        #[cfg(any())]
        panic!();
        
        #[cfg(not(any()))]
        return None;
    };
    Some(x)
}
```

I saw this happen:

```
  --> src/lib.rs:6:5
   |
6  | /     let Some(x) = returns_option() else {
7  | |         #[cfg(any())]
8  | |         panic!();
...  |
11 | |         return None;
12 | |     };
   | |______^ help: replace it with: `let x = returns_option()?;`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#question_mark
   = note: `#[warn(clippy::question_mark)]` on by default
```

I expected to see this happen: (no output).

### Version

rustc 1.79.0 (129f3b996 2024-06-10)
binary: rustc
commit-hash: 129f3b9964af4d4a709d1383930ade12dfe7c081
commit-date: 2024-06-10
host: x86_64-unknown-linux-gnu
release: 1.79.0
LLVM version: 18.1.7

### Additional Labels

@rustbot label +I-suggestion-causes-bug

**Comment by GoldsteinE** [comment]
Weirdly, adding `#[allow(clippy::question_mark)]` before `let` doesn’t make the warning go away.

**Comment by est31** [comment]
> Weirdly, adding `#[allow(clippy::question_mark)]` before `let` doesn’t make the warning go away.

That's also a bug. As the lint is emitted in the `manual_let_else` code, adding `#[allow(clippy::manual_let_else)]` works, even though it shouldn't.

**Comment by blyxyas** [comment]
[`span_contains_cfg`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/attrs/fn.span_contains_cfg.html) is done just for this kind of situation.

Currently, `clippy --fix` doesn't try to fix it, it's a human-reviewed suggestion (at least) Although I agree that we should either add a note to the lint message or don't lint altogether.

**Comment by GoldsteinE** [comment]
Thanks for the guidance! I opened a PR to suppress the lint.
